File: examples/nodejs/_6_FixedString/publisher.js
const SharedPubSub = require('SharedPubSub');
const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

const publisher = new SharedPubSub.Publisher_FixedString2048("FixedString");

async function loop() {
    while (true) {
        /*--------- Example 1 : Publish ---------*/
        let value = "EXAMPLE 1";
        publisher.publish(value);
        console.log(`PUBLISHER JS : ${publisher.readValue()} Normal publish`)
        await sleep(1000);

        /*--------- Example 2 : Publish only on change ---------*/
        for(let i=0;i<100;++i){
            publisher.publishOnChange("EXAMPLE 1");
        }
        publisher.publishOnChange("EXAMPLE 2");
        console.log(`PUBLISHER JS : ${publisher.readValue()} Published on change`)
        await sleep(1000);

        /*--------- Example 3 : Push multiple in queue and notify after ---------*/
        for (let i = 0; i < 10; i++) {
            value += 1;
            publisher.setValueAndPush(`EXAMPLE ${i}`);
            console.log(`PUBLISHER JS : ${publisher.readValue()}`);
        }

        publisher.notifyAll();
        console.log(`PUBLISHER JS : ${publisher.readValue()} Sent Multiple`);
        await sleep(1000);
    }
}

loop().catch(console.error);

File: examples/python/_6_FixedString/publisher.py
from SharedPubSub import *
from time import sleep

publisher = Publisher_FixedString2048("FixedString")
value = FixedString2048()

while(True):

    ### Example 1 : Publish ###
    value.set("EXAMPLE 1")
    publisher.publish(value)
    print("PUBLISHER PY :", value.get(), "Normal publish")
    sleep(1)

    ### Example 2 : Publish only on change ###

    for i in range(0,100):
        publisher.publishOnChange(value)

    value.set("EXAMPLE 2")
    publisher.publishOnChange(value)
    print("PUBLISHER PY :", value.get(), "Published on change")
    sleep(1)

    ### Example 3 : Push multiple in queue and notify after ###
    for i in range(0,10):
        value.set("EXAMPLE 3 "+str(i))
        publisher.setValueAndPush(value)
        print("PUBLISHER PY :", value.get())
    
    publisher.notifyAll()
    print("PUBLISHER PY :", value.get(),"Sent Multiple")
    sleep(1)